generators/typescript-v2/ast/src/ast/Function.ts

```diff
@@ -0,0 +1,89 @@
+import { CodeBlock } from "./CodeBlock";
+import { Comment } from "./Comment";
+import { Parameter } from "./Parameter";
+import { Type } from "./Type";
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+
+export declare namespace Function {
+    interface Args {
+        name: string;
+        parameters: Parameter[];
+        return_: Type[];
```

Comment:
Return types in TS should be a single value, not an array IMO.
If we need multiple values it should be a single Type, and the type in question would be an array or something similar.

Reply:
Yea, this is probably a relic of the Go AST, which often requires multiple return args (for `error`).